### PR TITLE
[FLINK-22452][docs] list the configuration methods of FlinkKafkaProducer and add ProducerFencedException in Troubleshooting

### PR DESCRIPTION
**Pull request by predatorray**
## What is the purpose of the change

Adding more documentation for FlinkKafkaProducer, especially for the `setTransactionalIdPrefix()` method.

This change code is the follow-up of #16494, which adds missing documentation for that feature.

## Brief change log

  - added a list of additional things that could be configured for `FlinkKafkaProducer`
  - added `ProducerFencedException` under the Troubleshooting section to explain the situation where `setTransactionalIdPrefix()` should be used

## Verifying this change

This change is a trivial rework of documentation without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: docs/content.zh/docs/connectors/datastream/kafka.md [hunk 1]
```diff
@@ -389,6 +389,13 @@ stream.addSink(myProducer)
 {{< /tab >}}
 {{< /tabs >}}
 
+除此之外，我们还可以使用下列配置方法：
+
+- {{< javadoc name="setWriteTimestampToKafka(boolean writeTimestampToKafka)" file="horg/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.html#setWriteTimestampToKafka-boolean-" >}}，给每条记录设置时间戳
```
Comment (pnowojski) [inline docs/content.zh/docs/connectors/datastream/kafka.md:394]: `horg`?

**Review by pnowojski — APPROVED** [review]
LGTM % probably one typo

File: docs/content.zh/docs/connectors/datastream/kafka.md [hunk 2]
```diff
@@ -391,7 +391,8 @@ stream.addSink(myProducer)
 
 除此之外，我们还可以使用下列配置方法：
 
-- {{< javadoc name="setWriteTimestampToKafka(boolean writeTimestampToKafka)" file="horg/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.html#setWriteTimestampToKafka-boolean-" >}}，给每条记录设置时间戳
+- {{< javadoc name="set
```
Comment (pnowojski) [inline docs/content.zh/docs/connectors/datastream/kafka.md:394]: This new line is failing the azure build unfortunately (and we need green build to be allowed to merge a PR). Could you fix it and squash already all commits?

Comment (predatorray) [reply docs/content.zh/docs/connectors/datastream/kafka.md:394]: Sure. Let me fix it.

**Review by pnowojski — APPROVED** [review]
Thanks for the contribution and translation to Chinese! :)

**Comment by predatorray** [comment]
@pnowojski Thanks for your review!

**Pull request merged**